### ISSUE TITLE
headless Emulator exiting after starting with no error message

### ISSUE THREAD

**Issue by nkshschdv** [amrsa1/Android-Emulator-image#12]
Command to build docker image 
Steps: 
1. Build an image 
$ docker build --build-arg ARCH=x86_64 --build-arg TARGET=google_apis_playstore --build-arg API_LEVEL=31 --build-arg BUILD_TOOLS=31.0.0 --build-arg EMULATOR_DEVICE="Nexus 6" --build-arg EMULATOR_NAME=nexus -t my-android-image .  
2. I have to perform this command as windows OS was adding it's own characters after cloning to bash files seems 
$ docker exec --privileged -it -e EMULATOR_TIMEOUT=300 androidContainer bash -c "sed -i -e 's/\r$//' start_emu_headless.sh"
3. Start the emulator 
$ docker exec --privileged -it androidContainer /bin/dash -c "./start_emu_headless.sh" 

![image](https://github.com/amrsa1/Android-Emulator-image/assets/15321028/63a73c17-b004-4516-8f1a-9f1e0c65807d)

4. however after i check the emulator status with below command 
$ docker exec --privileged -it androidContainer /bin/dash -c "adb devices "

**Comment by nkshschdv** [comment]
So funny to answer my own question , but i found the solution jsut now , and it is working as expected. here https://stackoverflow.com/questions/42218957/dockerfile-cmd-instruction-will-exit-the-container-just-after-running-it  


Changed the emulator start command with the following 
$ docker exec --privileged -it androidContainer /bin/dash -c "./start_emu_headless.sh ; sleep infinity"

**Comment by amrsa1** [comment]
> Command to build docker image 
> Steps: 
> 1. Build an image 
> $ docker build --build-arg ARCH=x86_64 --build-arg TARGET=google_apis_playstore --build-arg API_LEVEL=31 --build-arg BUILD_TOOLS=31.0.0 --build-arg EMULATOR_DEVICE="Nexus 6" --build-arg EMULATOR_NAME=nexus -t my-android-image .  
> 2. I have to perform this command as windows OS was adding it's own characters after cloning to bash files seems 
> $ docker exec --privileged -it -e EMULATOR_TIMEOUT=300 androidContainer bash -c "sed -i -e 's/\r$//' start_emu_headless.sh"
> 3. Start the emulator 
> $ docker exec --privileged -it androidContainer /bin/dash -c "./start_emu_headless.sh" 
> 
> ![image](https://github.com/amrsa1/Android-Emulator-image/assets/15321028/63a73c17-b004-4516-8f1a-9f1e0c65807d)
> 
> 4. however after i check the emulator status with below command , it shows that it has no devices attached
> $ docker exec --privileged -it androidContainer /bin/dash -c "adb devices " 
> 
> ![image](https://github.com/amrsa1/Android-Emulator-image/assets/15321028/23195b77-f4d7-4e06-9940-b8066937b3d6)
> 
> 
> Why the devices start but exit just after executing ? What is the way to persist it ?
> 

Does it exit and terminate the container or just exiting ?

**Comment by nkshschdv** [comment]
> > Command to build docker image
> > Steps:
> > 
> > 1. Build an image
> >    $ docker build --build-arg ARCH=x86_64 --build-arg TARGET=google_apis_playstore --build-arg API_LEVEL=31 --build-arg BUILD_TOOLS=31.0.0 --build-arg EMULATOR_DEVICE="Nexus 6" --build-arg EMULATOR_NAME=nexus -t my-android-image .
> > 2. I have to perform this command as windows OS was adding it's own characters after cloning to bash files seems
> >    $ docker exec --privileged -it -e EMULATOR_TIMEOUT=300 androidContainer bash -c "sed -i -e 's/\r$//' start_emu_headless.sh"
> > 3. Start the emulator
> >    $ docker exec --privileged -it androidContainer /bin/dash -c "./start_emu_headless.sh"
> > 
> > ![image](https://user-images.githubusercontent.com/15321028/284865897-63a73c17-b004-4516-8f1a-9f1e0c65807d.png)
> > 
> > 4. however after i check the emulator status with below command , it shows that it has no devices attached
> >    $ docker exec --privileged -it androidContainer /bin/dash -c "adb devices "
> > 
> > ![image](https://user-images.githubusercontent.com/15321028/284866277-23195b77-f4d7-4e06-9940-b8066937b3d6.png)
> > Why the devices start but exit just after executing ? What is the way to persist it ?
> 
> Does it exit and terminate the container or just exiting ?

No, it's not exiting the container it still show running in docker ps command . I missed to mention in my question , how i run the container. I added that as well.

**Comment by nkshschdv** [comment]
I'm assuming this is not happeniing to you @amrsa1.  You are using which OS ?
